pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_conversion.go

```diff
@@ -138,34 +136,26 @@ func k8sToKumaHeader(header gatewayapi.HTTPHeader) *mesh_proto.GatewayRoute_Http
 func (r *HTTPRouteReconciler) gapiToKumaRef(
 	ctx context.Context, mesh string, objectNamespace string, ref gatewayapi.BackendObjectReference,
 ) (map[string]string, *kube_meta.Condition, error) {
-	name := string(ref.Name)
-	// Group and Kind both have default values
-	group := string(*ref.Group)
-	kind := string(*ref.Kind)
-
-	namespace := objectNamespace
-	if ref.Namespace != nil {
-		namespace = string(*ref.Namespace)
-	}
-
 	policyRef := policy.PolicyReferenceBackend(policy.FromHTTPRouteIn(objectNamespace), ref)
 
+	gk := policyRef.GroupKindReferredTo()
+	namespacedName := policyRef.NamespacedNameReferredTo()
+
 	if permitted, err := policy.IsReferencePermitted(ctx, r.Client, policyRef); err != nil {
 		return nil, nil, errors.Wrap(err, "couldn't determine if backend reference is permitted")
 	} else if !permitted {
-		targetGK := kube_schema.GroupKind{Group: string(*ref.Group), Kind: string(*ref.Kind)}
 		return nil,
 			&kube_meta.Condition{
 				Type:    string(gatewayapi.ConditionRouteResolvedRefs),
 				Status:  kube_meta.ConditionFalse,
 				Reason:  RefNotPermitted,
-				Message: fmt.Sprintf("reference to %s %s/%s not permitted by any ReferencePolicy", targetGK.String(), policyRef.ToNamespace(), ref.Name),
+				Message: fmt.Sprintf("reference to %s %s not permitted by any ReferencePolicy", gk, namespacedName),
```

Comment:
Nit: I wanna suggest using %q for the NamespacedName, but it's borderline, so if you don't agree, don't worry about it :)